src/I18n/Parser/PoFileParser.php

```diff
@@ -142,11 +142,10 @@ protected function _addMessage(array &$messages, array $item)
 
         $translation = stripcslashes($translation);
 
-        if ($context && (!isset($messages[$singular]) || is_array($messages[$singular]))) {
+        if ($context !== null) {
             $messages[$singular]['_context'][$context] = $translation;
-        }
-        if ($context === null) {
-            $messages[$singular] = $translation;
+        } else {
+            $messages[$singular]['_context'][''] = $translation;
```

Comment:
This change adds an extra array and extra string for every translated value. The delta looks like it is 424 bytes per message though. In an application with 10K messages we'd be adding ~4.1MB

```
$ php -r '$a = "testing"; echo memory_get_usage() . "\n\n";'
350064

$ php -r '$a = ["" => "testing"]; echo memory_get_usage() . "\n\n";'
350488
```